AmplifyTools/amplify-tools.sh

```diff
@@ -8,11 +8,45 @@
 set -e
 export PATH=$PATH:$(npm bin -g)
 
+MIN_SUPPORTED_VERSION_MAJOR=2
+MIN_SUPPORTED_VERSION_MINOR=17
+MIN_SUPPORTED_VERSION_PATCH=1
+
+CURR_VERSION=`npx -q amplify-app --version`
+CURR_VERSION_MAJOR=`echo ${CURR_VERSION} | tr  '.' ' ' | awk '{print $1}'`
+CURR_VERSION_MINOR=`echo ${CURR_VERSION} | tr  '.' ' ' | awk '{print $2}'`
+CURR_VERSION_PATCH=`echo ${CURR_VERSION} | tr  '.' ' ' | awk '{print $3}'`
+LOCAL_AMPAPP_VERSION_INVALID=0
+
+checkVersionAmplifyApp() {
+    if [ ${CURR_VERSION_MAJOR} -lt ${MIN_SUPPORTED_VERSION_MAJOR} ]; then
+	LOCAL_AMPAPP_VERSION_INVALID=1
+	return
+    else
+	if [ ${CURR_VERSION_MINOR} -lt ${MIN_SUPPORTED_VERSION_MINOR} ]; then
+	    LOCAL_AMPAPP_VERSION_INVALID=1
+	    return
+	else
+	    if [ ${CURR_VERSION_PATCH} -lt ${MIN_SUPPORTED_VERSION_PATCH} ]; then
+		LOCAL_AMPAPP_VERSION_INVALID=1
+		return
+	    fi
+	fi
+    fi
+}
+checkVersionAmplifyApp
```

Comment:
Need to do some sanity checking around inputs before calling this function... otherwise, we can run into errors.